bento/crates/workflow/src/tasks/resolve.rs

```diff
@@ -52,25 +92,29 @@ pub async fn resolver(agent: &Agent, job_id: &Uuid, request: &ResolveReq) -> Res
                         .prover
                         .as_ref()
                         .context("Missing prover from resolve task")?
-                        .resolve(&conditional_receipt, &assumption_receipt)?;
+                        .resolve(&conditional_receipt, &assumption_receipt)
+                        .context("Failed to resolve the conditional receipt")?;
                 }
                 tracing::info!("Resolve complete");
             }
         }
     }
 
     // Write out the resolved receipt
+    tracing::info!("Serializing resolved receipt");
     let serialized_asset =
-        serialize_obj(&conditional_receipt).expect("Failed to serialize the asset");
+        serialize_obj(&conditional_receipt).context("Failed to serialize resolved receipt")?;
 
+    tracing::info!("Writing resolved receipt to Redis key: {root_receipt_key}");
     redis::set_key_with_expiry(
         &mut conn,
         &root_receipt_key,
         serialized_asset,
         Some(agent.args.redis_ttl),
     )
     .await
-    .expect("Failed to set root receipt key with expiry");
+    .context("Failed to set root receipt key with expiry")?;
 
+    tracing::info!("Resolve operation completed successfully");
```

Comment:
perhaps the logs in changes should be debug over info? Can change in a future change (logs are very noisy right now)